### PR TITLE
Multiple inheritance

### PR DESCRIPTION
**Pull request by kunjsong01**
This patch enables the multi-inheritance test case inheritance01 and inheritance01_fail.  

Fixed member function call to enable this test case.

Error message showing ESBMC failed to cope with method call: 
```
Generating GOTO Program
ERROR: unexpected function argument: member
Aborted (core dumped)
```

**Before vs. After**
Symbole table: 
Screenshot 2023-01-30 at 16 50 34

7 more test cases have been enabled after fixing this error: 

regression/esbmc-cpp/inheritance_bringup/inheritance01
regression/esbmc-cpp/inheritance_bringup/inheritance01_fail
regression/esbmc-cpp11/constructors/Constructor5-1
regression/esbmc-cpp11/reference/Reference1
regression/esbmc-cpp11/reference/Reference1-fail
regression/esbmc-cpp11/reference/Reference2
regression/esbmc-cpp11/reference/Reference2-fail

Also enables the following test cases added in https://github.com/esbmc/esbmc/pull/890: 
regression/esbmc-cpp/inheritance_bringup/dynamic_cast4 
regression/esbmc-cpp/inheritance_bringup/dynamic_cast4_fail 
regression/esbmc-cpp/inheritance_bringup/inheritance07 
regression/esbmc-cpp/inheritance_bringup/inheritance07_fail

### CODE REVIEW

File: src/clang-c-frontend/clang_c_convert.cpp [hunk 1]
```diff
@@ -298,13 +303,22 @@ bool clang_c_convertert::get_struct_union_class(const clang::RecordDecl &rd)
   if(context.find_symbol(id) != nullptr)
     return false;
 
+  // TODO: Fix me when we have a test case using C++ unioin.
```
Comment (lucasccordeiro) [inline src/clang-c-frontend/clang_c_convert.cpp]: "unioin" -> "union"

File: src/clang-c-frontend/clang_c_convert.cpp [hunk 2]
```diff
@@ -3291,3 +3280,51 @@ clang_c_convertert::get_top_FunctionDecl_from_Stmt(const clang::Stmt &stmt)
...
+bool clang_c_convertert::check_alignment_attributes(
+  const clang::FieldDecl *field,
+  struct_typet::componentt &comp)
+{
+  if(field->hasAttrs())
+  {
+    const auto &attrs = field->getAttrs();
+    for(const auto &attr : attrs)
+    {
+      if(attr->getKind() == clang::attr::Aligned)
```
Comment (lucasccordeiro) [inline src/clang-c-frontend/clang_c_convert.cpp]: Add comments for each case.

File: src/util/show_symbol_table.cpp [hunk 3]
```diff
@@ -33,10 +33,20 @@ void show_symbol_table_plain(const namespacet &ns, std::ostream &out)
     std::string type_str, value_str;
 
     if(s.type.is_not_nil())
+    {
```
Comment (lucasccordeiro) [inline src/util/show_symbol_table.cpp]: Shall we use our log function to report these messages?

**Review by lucasccordeiro — CHANGES_REQUESTED** [review]
Minor comments.

**Review by lucasccordeiro — APPROVED** [review]
LGTM.

**Review by rafaelsamenezes — APPROVED** [review]

**Review by shmarovfedor — APPROVED** [review]

**Pull request merged**